apps/sunrise/app.js

```diff
@@ -14,333 +13,206 @@ function loadLocation () {
-
-  time = localMeanTime - (longitude / Date.DEGREES_PER_HOUR);
-  time = Math.mod(time, 24);
-
-  const midnight = new Date(0);
-  // midnight.setUTCFullYear(this.getUTCFullYear());
-  // midnight.setUTCMonth(this.getUTCMonth());
-  // midnight.setUTCDate(this.getUTCDate());
-
-  const milli = midnight.getTime() + (time * 60 * 60 * 1000);
-
-  return new Date(milli);
-};
-
-Date.DEGREES_PER_HOUR = 360 / 24;
-
-// Utility functions
-
-Date.prototype.getDayOfYear = function () {
-  const onejan = new Date(this.getFullYear(), 0, 1);
-  return Math.ceil((this - onejan) / 86400000);
-};
-
-Math.degToRad = function (num) {
-  return num * Math.PI / 180;
-};
-
-Math.radToDeg = function (radians) {
-  return radians * 180.0 / Math.PI;
-};
-
-Math.sinDeg = function (deg) {
-  return Math.sin(deg * 2.0 * Math.PI / 360.0);
-};
-
-Math.acosDeg = function (x) {
-  return Math.acos(x) * 360.0 / (2 * Math.PI);
-};
-
-Math.asinDeg = function (x) {
-  return Math.asin(x) * 360.0 / (2 * Math.PI);
-};
-
-Math.tanDeg = function (deg) {
-  return Math.tan(deg * 2.0 * Math.PI / 360.0);
-};
-
-Math.cosDeg = function (deg) {
-  return Math.cos(deg * 2.0 * Math.PI / 360.0);
-};
-
-Math.mod = function (a, b) {
-  let result = a % b;
-  if (result < 0) {
-    result += b;
-  }
-  return result;
-};
-
-const sunrise = new Date().sunrise(lat, lon);
-const sunset = new Date().sunset(lat, lon);
-
 const w = g.getWidth();
 const h = g.getHeight();
-const oy = h / 1.7;
 
-let sunRiseX = 0;
-let sunSetX = 0;
-const sinStep = 13;
+let sunRiseX;
+let sunSetX;
+let solarNoonX;
 
 let pos = 0;
 let realTime = true;
 const r = 10;
 
 let frames = 0; // amount of pending frames to render (0 if none)
-// set to 1 because pos 0 is displayed as 0-1:59
-let curPos = 1; // x position of the sun
+let curPos = 0; // x position of the sun
 let realPos = 0; // x position of the sun depending on currentime
 
+let day;
+let now = new Date();
+
+let sr; // sunrise formatted as time
+let ss; // sunset formatted as time
+
+let slope; // slope for sea leve line
+let yint; // y-intercept for sea leve line
+
+const Locale = require("locale");
+const SunCalc = require("suncalc");
+
+let alt = 0; // altitude
+
+let daystart = new Date();
+daystart.setHours(0);
+daystart.setMinutes(0);
+daystart.setSeconds(0);
 
-function formatAsTime (hour, minute) {
-  return '' + ((hour < 10) ? '0' : '') + (0 | hour) +
-         ':' + ((minute < 10) ? '0' : '') + (0 | minute);
+function getAltitude () {
+  const p = Bangle.getPressure();
+  if (p) {
+    p.then(d => {
+      if (d) alt = d.altitude;
+    });
+  }
 }
 
 function drawSinuses () {
-  let x = 0;
-
+  // messy because 1d array for 2 axes
+  const array = [];
+  const sinStep = 12;
+  let i;
   g.setColor(1, 1, 1);
-  let y = ypos(x);
-  while (x < w) {
-    y2 = ypos(x + sinStep);
-    g.drawLine(x, y, x + sinStep, y2);
-    y = y2;
-    x += sinStep; // no need to draw all steps
+
+  for (i = 0; i <= w - sinStep; i += sinStep) {
```

Comment:
The sine function can look crooked at the right side of the screen.

It should be like this instead:
`for (i = 0; i <= w; i += sinStep) {`

The 'w - sinStep' made sense before, but not anymore.

Reply:
Ah thank you - [sorted that now](https://github.com/espruino/BangleApps/pull/3103/commits/4bdc61c6376ad66d52d45659f6c6b61deacac753)